### DISCUSSION TITLE
[Question] How much data able to be saved on Iron Session?

### DISCUSSION THREAD

**Discussion by benyaminl** [vvo/iron-session#645] [Q&A] [answered]
This is probably a dumb question,  but I'm just want to make sure, that the cookie is containing a function right, and cookie maximum data is around 4kb, so much much data in theory iron session could save? Is it lower than 1-2kb? Thank you

**Comment by vvo** [comment, marked as answer]
Hey @benyaminl you'd have to do the testing yourself. But I would expect 3KB of data to be acceptable. Most of the time I only use it to store 10 properties maximum. What are you trying to store that is bigger than 1-2KB? Thanks!

**Reply by benyaminl** [reply]
No, I don't think I will save more than 2kbs tbh. It's just a new concept for me, as "Session", where I come from (.NET, PHP, Java), can have unlimited storage (as ram and disk available in server), so that's why I tried to compare them, even I know that Iron Session is Cookie based and stateless (well with cookie on client, at least we can have two identic container without loosing the data, as it's on cookie..). 

Thanks for clarifying. sorry for dumb question.

**Reply by vvo** [reply]
Not a dumb question at all 👍

**Comment by vvo** [comment]
Real numbers now, since v9 measures the cookie instead of estimating it: a browser refuses a cookie over 4096 bytes, and iron-session throws rather than let one be dropped silently. Encryption adds overhead, so plan for about 3KB of actual data.

If you need more, v9 has `chunk: true`, which splits the session across up to 4 cookies. Know the real ceiling before turning it on. Every cookie is sent on every request, nginx caps the whole `Cookie` header at 8KB by default, and a CDN in front of it may allow less. Over that you get a 400 or 431 at the edge, before your code runs.

For anything big, keep an id in the session and the data in your database. More detail: https://github.com/vvo/iron-session#session-size
